src/plots/plots.js

```diff
@@ -540,6 +544,23 @@ plots.supplyDefaults = function(gd) {
             (gd.calcdata[i][0] || {}).trace = trace;
         }
     }
+
+    // Set up the default keyframe if it doesn't exist:
+    if(!gd._frameData) {
+        gd._frameData = {};
+    }
+
+    if(!gd._frameData._frames) {
+        gd._frameData._frames = [];
+    }
+
+    if(!gd._frameData._frameHash) {
+        gd._frameData._frameHash = {};
+    }
+
+    if(!gd._frameData._counter) {
+        gd._frameData._counter = 0;
+    }
```

Comment:
I guess it is convenient for `_frameData` to be defined regardless if frames are present or not.

I'd vote for initialising `_frameData` is a separate methods off `Plots` to be called below this [line](https://github.com/plotly/plotly.js/blob/master/src/plot_api/plot_api.js#L121) in `Plotly.plot`.